src/components/select/select.template.tsx

```diff
@@ -265,6 +267,23 @@ function createMenuItems(
         });
 }
 
+function getMenuOptionFilter(selectIsRequired) {
+    return (option: Option) => {
+        if (!selectIsRequired) {
+            // If the select component is NOT required, we keep all options.
+            return true;
+        }
+        if (option.text) {
+            // If the select component IS required, we keep only options
+            // that are not "empty". We only check the text property, because
+            // some systems use an "empty option" that will have a value for
+            // the `value` property, to signify "no option selected". Such
+            // an option should be treated as "empty".
+            return true;
+        }
+    };
+}
```

Comment:
I made this fixup, so that instead of checking if the option is disabled, we check only if the select component itself is set to `required`, and if it is, we remove any options that have no text value.

As you can see, I also refactored a little bit, to have space for some comments.